src/ProductConstructionService/ProductConstructionService.Api/VirtualMonoRepo/VmrConfiguration.cs

```diff
@@ -19,13 +19,13 @@ public static class VmrConfiguration
 
     public static void AddVmrRegistrations(this WebApplicationBuilder builder, string vmrPath, string tmpPath, string vmrUri)
     {
-        builder.Services.TryAddSingleton<IBasicBarClient, SqlBarClient>();
+        builder.Services.TryAddTransient<IBasicBarClient, SqlBarClient>();
         builder.Services.AddVmrManagers(
             "git",
             vmrPath,
             tmpPath,
-            builder.Configuration["BotAccount-dotnet-bot-repo-PAT"],
-            builder.Configuration["dn-bot-all-orgs-code-r"]);
+            builder.Configuration[PcsConfiguration.GitHubToken],
```

Comment:
we can use the `GetRequiredValue` extension method here, just to be safer

Reply:
Here, I am not sure. The VMR sync can work without tokens when everything is public. We might like that possibly?

Reply:
that's a good point. the `AddVmrManagers` can already deal with nulls